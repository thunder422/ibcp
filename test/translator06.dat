#####################################################
#  Translator Test #6: PRINT Commands (2010-06-01)  #
#####################################################

PRINT A;B%;C$
PRINT A;B%;C$;
PRINT A,B%,C$
PRINT A,B%,C$,
PRINT A+B; C%+D%; E$+F$; MID$(E$,1); LEFT$(E$+F$,5)
PRINT "Test:"; TAB(10); A%; SPC(5); B
PRINT "Test:"; TAB(10); A; LOG(5); B
PRINT A; TAB(10)
PRINT A; TAB(10);
PRINT A; SPC(10)
PRINT A; SPC(10);
PRINT ,,A
PRINT ,,A,
PRINT A,,B
PRINT A,,B,
PRINT A;;B
PRINT;A;;B;;
PRINT;
PRINT,

# PRINT error tests
PRINT C(A,,B)
PRINT C(A;B)
A=SPC(10)
A=TAB(10)
PRINT (A,,B)
PRINT A+;
PRINT A+,
PRINT A+
PRINT INT(A+;
PRINT INT(A+,
PRINT INT(A+
PRINT INT(B,,C)
PRINT (A+B
PRINT (A+B,
PRINT MID$(A$+B$

# 2011-03-03: more PRINT error tests
PRINT (TAB(10)
PRINT INT(TAB(10))
PRINT VAL(TAB(10))
PRINT A(TAB(10))
PRINT A+TAB(10)
PRINT A%+TAB(10)
PRINT A$+TAB(10)
PRINT TAB(10)+A
