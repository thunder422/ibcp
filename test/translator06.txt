
Input: PRINT A;B%;C$
Output: A PrintDbl B% PrintInt C$ PrintStr PRINT 

Input: PRINT A;B%;C$;
Output: A PrintDbl B% PrintInt C$ PrintStr ; 

Input: PRINT A,B%,C$
Output: A PrintDbl , B% PrintInt , C$ PrintStr PRINT 

Input: PRINT A,B%,C$,
Output: A PrintDbl , B% PrintInt , C$ PrintStr , PRINT 

Input: PRINT A+B; C%+D%; E$+F$; MID$(E$,1); LEFT$(E$+F$,5)
Output: A B + PrintDbl C% D% +% PrintInt E$ F$ +$ PrintStr E$ 1% MID2$( PrintStr E$ F$ +$ 5% LEFT$( PrintStr PRINT 

Input: PRINT "Test:"; TAB(10); A%; SPC(5); B
Output: "Test:" PrintStr 10% TAB( A% PrintInt 5% SPC( B PrintDbl PRINT 

Input: PRINT "Test:"; TAB(10); A; LOG(5); B
Output: "Test:" PrintStr 10% TAB( A PrintDbl 5 LOG( PrintDbl B PrintDbl PRINT 

Input: PRINT A; TAB(10)
Output: A PrintDbl 10% TAB( PRINT 

Input: PRINT A; TAB(10);
Output: A PrintDbl 10% TAB( ; 

Input: PRINT A; SPC(10)
Output: A PrintDbl 10% SPC( PRINT 

Input: PRINT A; SPC(10);
Output: A PrintDbl 10% SPC( ; 

Input: PRINT ,,A
Output: , , A PrintDbl PRINT 

Input: PRINT ,,A,
Output: , , A PrintDbl , PRINT 

Input: PRINT A,,B
Output: A PrintDbl , , B PrintDbl PRINT 

Input: PRINT A,,B,
Output: A PrintDbl , , B PrintDbl , PRINT 

Input: PRINT A;;B
               ^-- expected expression, print function or end-of-statement

Input: PRINT;A;;B;;
            ^-- expected expression, comma, print function or end-of-statement

Input: PRINT;
            ^-- expected expression, comma, print function or end-of-statement

Input: PRINT,
Output: , PRINT 

Input: PRINT C(A,,B)
                 ^-- expected expression

Input: PRINT C(A;B)
                ^-- expected operator, comma or closing parentheses

Input: A=SPC(10)
         ^^^^-- expected numeric expression

Input: A=TAB(10)
         ^^^^-- expected numeric expression

Input: PRINT (A,,B)
               ^-- expected operator or closing parentheses

Input: PRINT A+;
               ^-- expected numeric expression

Input: PRINT A+,
               ^-- expected numeric expression

Input: PRINT A+
               ^-- expected numeric expression

Input: PRINT INT(A+;
                   ^-- expected numeric expression

Input: PRINT INT(A+,
                   ^-- expected numeric expression

Input: PRINT INT(A+
                   ^-- expected numeric expression

Input: PRINT INT(B,,C)
                  ^-- expected operator or closing parentheses

Input: PRINT (A+B
                 ^-- expected operator or closing parentheses

Input: PRINT (A+B,
                 ^-- expected operator or closing parentheses

Input: PRINT MID$(A$+B$
                       ^-- expected operator or comma

Input: PRINT (TAB(10)
              ^^^^-- expected expression

Input: PRINT INT(TAB(10))
                 ^^^^-- expected numeric expression

Input: PRINT VAL(TAB(10))
                 ^^^^-- expected string expression

Input: PRINT A(TAB(10))
               ^^^^-- expected expression

Input: PRINT A+TAB(10)
               ^^^^-- expected numeric expression

Input: PRINT A%+TAB(10)
                ^^^^-- expected numeric expression

Input: PRINT A$+TAB(10)
                ^^^^-- expected string expression

Input: PRINT TAB(10)+A
                    ^-- expected semicolon, comma or end-of-statement

Input: PRINT C(D(A,,B),E)
                   ^-- expected expression

Input: PRINT INT(D(A,,B),E)
                     ^-- expected expression

Input: PRINT A;,
               ^-- expected expression, print function or end-of-statement

