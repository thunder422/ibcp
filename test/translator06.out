
Input: PRINT A;B%;C$
Output: PRINT A; B%; C$ 

Input: PRINT A;B%;C$;
Output: PRINT A; B%; C$; 

Input: PRINT A,B%,C$
Output: PRINT A, B%, C$ 

Input: PRINT A,B%,C$,
Output: PRINT A, B%, C$, 

Input: PRINT A+B; C%+D%; E$+F$; MID$(E$,1); LEFT$(E$+F$,5)
Output: PRINT A + B; C% + D%; E$ + F$; MID$(E$, 1); LEFT$(E$ + F$, 5) 

Input: PRINT "Test:"; TAB(10); A%; SPC(5); B
Output: PRINT "Test:"; TAB(10); A%; SPC(5); B 

Input: PRINT "Test:"; TAB(10); A; LOG(5); B
Output: PRINT "Test:"; TAB(10); A; LOG(5); B 

Input: PRINT A; TAB(10)
Output: PRINT A; TAB(10) 

Input: PRINT A; TAB(10);
Output: PRINT A; TAB(10); 

Input: PRINT A; SPC(10)
Output: PRINT A; SPC(10) 

Input: PRINT A; SPC(10);
Output: PRINT A; SPC(10); 

Input: PRINT ,,A
Output: PRINT ,, A 

Input: PRINT ,,A,
Output: PRINT ,, A, 

Input: PRINT A,,B
Output: PRINT A,, B 

Input: PRINT A,,B,
Output: PRINT A,, B, 

Input: PRINT A;;B
               ^-- expected expression, print function or end-of-statement

Input: PRINT;A;;B;;
            ^-- expected expression, comma, print function or end-of-statement

Input: PRINT;
            ^-- expected expression, comma, print function or end-of-statement

Input: PRINT,
Output: PRINT , 

Input: PRINT C(A,,B)
                 ^-- expected numeric expression

Input: PRINT C(A;B)
                ^-- expected operator, comma or closing parentheses

Input: A=SPC(10)
         ^^^^-- expected numeric expression

Input: A=TAB(10)
         ^^^^-- expected numeric expression

Input: PRINT (A,,B)
               ^-- expected operator or closing parentheses

Input: PRINT A+;
               ^-- expected numeric expression

Input: PRINT A+,
               ^-- expected numeric expression

Input: PRINT A+
               ^-- expected numeric expression

Input: PRINT INT(A+;
                   ^-- expected numeric expression

Input: PRINT INT(A+,
                   ^-- expected numeric expression

Input: PRINT INT(A+
                   ^-- expected numeric expression

Input: PRINT INT(B,,C)
                  ^-- expected operator or closing parentheses

Input: PRINT (A+B
                 ^-- expected operator or closing parentheses

Input: PRINT (A+B,
                 ^-- expected operator or closing parentheses

Input: PRINT MID$(A$+B$
                       ^-- expected operator or comma

Input: PRINT (TAB(10)
              ^^^^-- expected expression

Input: PRINT INT(TAB(10))
                 ^^^^-- expected numeric expression

Input: PRINT VAL(TAB(10))
                 ^^^^-- expected string expression

Input: PRINT A(TAB(10))
               ^^^^-- expected numeric expression

Input: PRINT A+TAB(10)
               ^^^^-- expected numeric expression

Input: PRINT A%+TAB(10)
                ^^^^-- expected numeric expression

Input: PRINT A$+TAB(10)
                ^^^^-- expected string expression

Input: PRINT TAB(10)+A
                    ^-- expected semicolon, comma or end-of-statement

Input: PRINT C(D(A,,B),E)
                   ^-- expected numeric expression

Input: PRINT INT(D(A,,B),E)
                     ^-- expected numeric expression

Input: PRINT A;,
               ^-- expected expression, print function or end-of-statement

